lib/customisation.nix

```diff
@@ -269,10 +269,11 @@ rec {
     let self = f self // {
           newScope = scope: newScope (self // scope);
           callPackage = self.newScope {};
-          overrideScope = g: lib.warn
-            "`overrideScope` (from `lib.makeScope`) is deprecated. Do `overrideScope' (self: super: { … })` instead of `overrideScope (super: self: { … })`. All other overrides have the parameters in that order, including other definitions of `overrideScope`. This was the only definition violating the pattern."
-            (makeScope newScope (lib.fixedPoints.extends (lib.flip g) f));
-          overrideScope' = g: makeScope newScope (lib.fixedPoints.extends g f);
+          overrideScope = g: makeScope newScope (lib.fixedPoints.extends g f);
+          # Remove after 24.05 is released.
+          overrideScope' = g: lib.warn
+            "`overrideScope'` (from `lib.makeScope`) has been renamed to `overrideScope`."
```

Comment:
I needs to use `warnIf (isInOldestRelease 2311)`, such that we only warn once 23.11 isn't supported anymore, otherwise we still have the problem from https://github.com/NixOS/nixpkgs/pull/248988#discussion_r1292940144, because there wouldn't be a transition period where users could use both versions of the function.

Reply:
https://github.com/NixOS/nixpkgs/pull/248988#discussion_r1292984930

> My plan now is to backport the first commit to 23.05 so users can support stable and unstable at the same time

Reply:
I don't think that's a great idea, because then you can't just check whether a release is 23.05 to figure out if `overrideScope` is fine to use, instead you have to make sure it's 23.05 after a specific commit/date.

Using `warnIf (isInOldestRelease ..)` without a backport would be my preference, it means 23.11 will be a clear boundary of when it changed.

And anyways, there's no need to hurry with this deprecation, let's take it slow

Reply:
Uses warnIf now